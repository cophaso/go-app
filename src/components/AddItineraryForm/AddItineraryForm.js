import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import ItinerariesApiService from '../../services/Itinerary-api-service';
import TokenService from '../../services/token-service'

export default class AddItineraryForm extends Component {
  static defaultProps = {
    onAddItinerarySuccess: () => {}
  }

  state = {
    error: null
  };

  handleSubmit = ev => {
    ev.preventDefault()
    const { title, start_date, end_date } = ev.target

    this.setState({ error: null })
    
    ItinerariesApiService.postItinerary({
      title: title.value,
      start_date: start_date.value,
      end_date: end_date.value,
      user_id: localStorage.getItem('user_id')
    })
      .then(itinerary => {
        title.value = ''
        this.props.onAddItinerarySuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
        TokenService.clearAuthToken();
        window.location.pathname = '/login';
      })
  }

  render() {
    const { error } = this.state
    let content
    if (error != null) {
      content = <p className='red'>{error.name}</p>
    }
    return (
      <form
      className='AddItineraryForm'
        onSubmit={this.handleSubmit}
      >
        <div role='alert'>
          {content}
        </div>
        <div className='title'>
          <label htmlFor='AddItineraryForm__title'>
            Title <Required />
          </label>
          <Input
            name='title'
            type='text'
            required
            id='AddItineraryForm__title'>
          </Input>
        </div>
        <div className='start_date'>
          <label htmlFor='AddItineraryForm__start_date'>
            Start Date: <Required />
          </label>
          <Input
            name='start_date'
            type='date'
            required
            id='AddItineraryForm__start_date'>
          </Input>
        </div>
        <div className='end_date'>
          <label htmlFor='AddItineraryForm__end_date'>
            End Date: <Required />
          </label>
          <Input
            name='end_date'
            type='date'
            required
            id='AddItineraryForm__end_date'>
          </Input>
        </div>
        <Button type='submit'>
          Add
        </Button>
      </form>
    )}
}