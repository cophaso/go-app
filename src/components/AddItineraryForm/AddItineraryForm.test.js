import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddItineraryForm from './AddItineraryForm';

describe(`AddItineraryForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.AddItineraryForm by default', () => {
    const wrapper = shallow(<AddItineraryForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the AddItineraryForm given props', () => {
    const wrapper = shallow(<AddItineraryForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
