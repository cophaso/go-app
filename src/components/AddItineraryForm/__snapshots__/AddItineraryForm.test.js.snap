// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItineraryForm component renders a form.AddItineraryForm by default 1`] = `
<form
  className="AddItineraryForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="title"
  >
    <label
      htmlFor="AddItineraryForm__title"
    >
      Title 
      <Required />
    </label>
    <Input
      id="AddItineraryForm__title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="start_date"
  >
    <label
      htmlFor="AddItineraryForm__start_date"
    >
      Start Date: 
      <Required />
    </label>
    <Input
      id="AddItineraryForm__start_date"
      name="start_date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="end_date"
  >
    <label
      htmlFor="AddItineraryForm__end_date"
    >
      End Date: 
      <Required />
    </label>
    <Input
      id="AddItineraryForm__end_date"
      name="end_date"
      required={true}
      type="date"
    />
  </div>
  <Button
    type="submit"
  >
    Add
  </Button>
</form>
`;

exports[`AddItineraryForm component renders the AddItineraryForm given props 1`] = `
<form
  className="AddItineraryForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="title"
  >
    <label
      htmlFor="AddItineraryForm__title"
    >
      Title 
      <Required />
    </label>
    <Input
      id="AddItineraryForm__title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="start_date"
  >
    <label
      htmlFor="AddItineraryForm__start_date"
    >
      Start Date: 
      <Required />
    </label>
    <Input
      id="AddItineraryForm__start_date"
      name="start_date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="end_date"
  >
    <label
      htmlFor="AddItineraryForm__end_date"
    >
      End Date: 
      <Required />
    </label>
    <Input
      id="AddItineraryForm__end_date"
      name="end_date"
      required={true}
      type="date"
    />
  </div>
  <Button
    type="submit"
  >
    Add
  </Button>
</form>
`;
